client/src/lib/data.ts

```diff
@@ -324,6 +325,7 @@ export async function createMark({
     body: JSON.stringify({
       calendarId,
       date,
+      day: new Date(date + 'T00:00').getDay(),
```

Comment:
I just read about why you need to add 'T00:00' since it's unexpected. What a mess this JS language is! I learned something today.